### PR TITLE
Monitoring dashboards: Support grouping of panels into rows

### PR DESCRIPTION
**Pull request by kyoto**
Group panels according to the rows specified in the dashboard
definition. Also allow expanding / collapsing of rows and support
defaulting to collapsed.

### CODE REVIEW

**Review by TheRealJon — CHANGES_REQUESTED** [review]
Looks like this might not be working as expected. I pulled your branch and took a look at the 
'Kubernetes / Compute Resources / Cluster' dashboard. Looks like all of the rows titled "Network" are showing up in individual collapsible sections instead being collected into one. I assume that the intention would be for panels with the same title to be aggregated?

image

Also, there is an issue with the key prop being passed  to each row:

```
react_devtools_backend.js:2430 Warning: Encountered two children with the same key, `35`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
    in div (created by PanelsRow)
    in div (created by PanelsRow)
    in PanelsRow (created by Board)
    in Board (created by MonitoringDashboardsPage_)
    in div (created by Dashboard)
    in Dashboard (created by MonitoringDashboardsPage_)
    in MonitoringDashboardsPage_ (created by ConnectFunction)
    in ConnectFunction
    in ErrorBoundary
    in Unknown (created by Context.Consumer)
    in Route (created by MonitoringUI)
    in Switch (created by MonitoringUI)
    in MonitoringUI (created by AsyncComponent)
    ...
```

File: frontend/public/components/monitoring/dashboards/index.tsx [hunk 1]
```diff
@@ -552,7 +583,22 @@ const MonitoringDashboardsPage_: React.FC<MonitoringDashboardsPageProps> = ({
...
+        rows.push(panel);
+      } else {
+        const row = rows[rows.length - 1];
+        if (_.isNil(row.panels)) {
+          row.panels = [];
+        }
+        row.panels.push(panel);
+      }
+    });
+  }
```
[the remark this answers is not in the document]
Comment (kyoto) [reply frontend/public/components/monitoring/dashboards/index.tsx]: Yes, I think this is a little better using `reduce` 👍

**Comment by kyoto** [comment]
Thanks @TheRealJon

I changed the `key` prop to use the row's panel `id`s, which should ensure uniqueness.

The repeating "Network" groups is actually correct behavior because that is what the dashboard definition says. (Grafana renders it in the same way.)

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Pull request merged**